[RLlib; docs] Docs do-over (new API stack): Remove special "new API stack" page (move some of its content to migration guide). (#49713)
ray-project/ray: doc/source/rllib/rllib-rlmodule.rst

Enabling the RLModule API in the AlgorithmConfig
------------------------------------------------

In the new API stack, activated by default, RLlib exclusively uses RLModules.

If you're working with a legacy config or want to migrate ``ModelV2`` or ``Policy`` classes to the
new API stack, see the :ref:`new API stack migration guide <rllib-new-api-stack-migration-guide>` for more information.

> If configured your :py:class:`~ray.rllib.algorithms.algorithm.Algorithm` to the old API stack, use the

Reviewer: ```suggestion
If you configured the :py:class:`~ray.rllib.algorithms.algorithm.Algorithm` to the old API stack, use the
```

    In case of :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule._forward_inference`,
    RLlib always makes the generated distributions from returned key ``action_dist_inputs`` deterministic first through
    the :py:meth:`~ray.rllib.models.distributions.Distribution.to_deterministic` utility before a possible action sample step.
    For example, RLlib reduces the sampling from a Categorical distribution to selecting the ``argmax``

>     actions from the distribution's logits or probabilities.

Reviewer: ```suggestion
    actions from the distribution logits or probabilities.
```

                    }

Never override the constructor (``__init__``), however, note that the
:py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` class's constructor requires the following arguments

> and also receives these properly when you call a spec's ``build()`` method:

Reviewer: ```suggestion
and also receives these properly when you call a spec ``build()`` method:
```

            :end-before: __sphinx_doc_encoder_end__

> To plug in the custom MultiRLModule, :ref:`from the first tab <rllib-rlmodule-guide-implementing-custom-multi-rl-modules>`,

Reviewer: ```suggestion
To plug in the :ref:`custom MultiRLModule <rllib-rlmodule-guide-implementing-custom-multi-rl-modules>` from the first tab,
```

into your algorithm's config, create a :py:class:`~ray.rllib.core.rl_module.multi_rl_module.MultiRLModuleSpec`
with the new class and its constructor settings. Also, create one :py:class:`~ray.rllib.core.rl_module.rl_module.RLModuleSpec`

> for each agent and the shared encoder RLModule, because RLlib requires their observation- and action spaces and their

Reviewer: ```suggestion
for each agent and the shared encoder RLModule, because RLlib requires their observation and action spaces and their
```

    from ray.rllib.algorithms.ppo import PPOConfig

    # Create a new Algorithm (with the changed module config: 32 units instead of the

>     # default 256; otherwise loading the state of ``module`` will fail due to a shape

Reviewer: ```suggestion
    # default 256; otherwise loading the state of ``module`` fails due to a shape
```

    ppo = config.build()

> Now you can load the saved RLModule state (from the preceding ``module.save_to_path()``) directly

Reviewer: ```suggestion
Now you can load the saved RLModule state from the preceding ``module.save_to_path()``, directly
```

> into the running Algorithm's RLModules. Note that all RLModules within the algorithm get updated, the ones

Reviewer: ```suggestion
into the running Algorithm RLModules. Note that all RLModules within the algorithm get updated, the ones
```
